### PR TITLE
Change travis config to use cmake. (see #256)

### PR DESCRIPTION
**Pull request by soroshsabz**
ITNOA

This PR want to resolve issue #256

### CODE REVIEW

**Comment by soroshsabz** [comment]
@fnc12 Did you know why below assertion failed in clang in ubuntu?

```
1: unit_tests: /home/travis/build/fnc12/sqlite_orm/tests/tests.cpp:2255: void testWideString(): Assertion `storage.get<Alphabet>(id).letters == expectedString' failed.
```

**Comment by fnc12** [comment]
looks like that it is some encoding problem

**Comment by soroshsabz** [comment]
> looks like that it is some encoding problem

@fnc12 I agree with you, but the problem is I don't know why :D , ok no problem I try to investigating a way that ignore this problem ;)

**Comment by fnc12** [comment]
try to print both strings to define which one is incorrect. Next print byte by byte every string. It will allow us to determine encoding details

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+# Copyright (c) 2012-2019 Sébastien Rombauts (sebastien.rombauts@gmail.com)
+#
+# Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
```
Comment (fnc12) [inline build.sh]: This header contains info about incorrect license. I think you need to remove it

**Comment by soroshsabz** [comment]
@fnc12 That's right this PR is WIP and I think need more work to prepare for merging :)

**Comment by soroshsabz** [comment]
Open [Microsoft VCPKG PR](https://github.com/Microsoft/vcpkg/pull/5915) to resolve problem on building in mac

**Comment by soroshsabz** [comment]
I finally pass two test of three, so I wait to [Microsoft VCPKG PR](https://github.com/Microsoft/vcpkg/pull/5915) to continue work for third test. :)

**Comment by soroshsabz** [comment]
@fnc12 if you have couple of free time please see [mac os gcc compile error](https://travis-ci.org/fnc12/sqlite_orm/jobs/514925479) , and let see to have any idea to resolve this? or I have to investigate the problem.

thanks

**Comment by fnc12** [comment]
Ok, I'll try to solve it

**Comment by fnc12** [comment]
I think we can solve compilation issue if we replace `static_if` with something more generic in this code
```c++
template<class F, class C>
            const F* get_object_field_pointer(const object_type &obj, C c) {
                const F *res = nullptr;
                this->for_each_column_with_field_type<F>([&res, &c, &obj, this](auto &col){
                    using namespace static_magic;
                    using column_type = typename std::remove_reference<decltype(col)>::type;
                    using member_pointer_t = typename column_type::member_pointer_t;
                    using getter_type = typename column_type::getter_type;
                    using setter_type = typename column_type::setter_type;
                    if(!res){
                        static_if<std::is_same<C, member_pointer_t>{}>([&res, &obj, &col, &c]{
                            if(compare_any(col.member_pointer, c)){
                                res = &(obj.*col.member_pointer);
                            }
                        })();
                    }
                    if(!res){
                        static_if<std::is_same<C, getter_type>{}>([&res, &obj, &col, &c]{
                            if(compare_any(col.getter, c)){
                                res = &((obj).*(col.getter))();
                            }
                        })();
                    }
                    if(!res){
                        static_if<std::is_same<C, setter_type>{}>([&res, &obj, &col, &c]{
                            if(compare_any(col.setter, c)){
                                res = &((obj).*(col.getter))();
                            }
                        })();
                    }
                });
                return res;
            }
```

**Comment by soroshsabz** [comment]
I think I found some info about it on [gcc](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68071)

**Comment by soroshsabz** [comment]
@fnc12 

> I think we can solve compilation issue if we replace static_if with something more generic in this code
>             template<class F, class C>
>             const F* get_object_field_pointer(const object_type &obj, C c) {
>                 const F *res = nullptr;
>                 this->for_each_column_with_field_type<F>([&res, &c, &obj, this](auto &col){
>                     using namespace static_magic;
>                     using column_type = typename std::remove_reference<decltype(col)>::type;
>                     using member_pointer_t = typename column_type::member_pointer_t;
>                     using getter_type = typename column_type::getter_type;
>                     using setter_type = typename column_type::setter_type;
>                     if(!res){
>                         static_if<std::is_same<C, member_pointer_t>{}>([&res, &obj, &col, &c]{
>                             if(compare_any(col.member_pointer, c)){
>                                 res = &(obj.*col.member_pointer);
>                             }
>                         })();
>                     }
>                     if(!res){
>                         static_if<std::is_same<C, getter_type>{}>([&res, &obj, &col, &c]{
>                             if(compare_any(col.getter, c)){
>                                 res = &((obj).*(col.getter))();
>                             }
>                         })();
>                     }
>                     if(!res){
>                         static_if<std::is_same<C, setter_type>{}>([&res, &obj, &col, &c]{
>                             if(compare_any(col.setter, c)){
>                                 res = &((obj).*(col.getter))();
>                             }
>                         })();
>                     }
>                 });
>                 return res;
>             }

@fnc12 I finally found workaround about this bug in gcc and I successfully green test :) [reslut of test](https://travis-ci.org/fnc12/sqlite_orm/builds/515201467)

So i just need to clean this code :)

**Comment by soroshsabz** [comment]
@fnc12 I resolve compilation issue on `gcc` in `mac` with clean way. and test now pass :)

File: build.sh (hunk 1 above)
Comment (soroshsabz) [reply build.sh]: Ok I remove it.

File: .travis.yml [hunk 2]
```diff
@@ -1,81 +1,103 @@
+# ITNOA
+# Copyright (c) 2012-2019 Sebastien Rombauts (sebastien.rombauts@gmail.com)
```
[the remark this answers is not in the document]
Comment (soroshsabz) [reply .travis.yml]: Ok I remove it :)

**Comment by fnc12** [comment]
This is very cool. `static_if` is nice but it exists until C++17 with `if constexpr` becomes available

**Comment by soroshsabz** [comment]
> This is very cool. static_if is nice but it exists until C++17 with if constexpr becomes available

Yes :) `if constexpr` is great, but backward compatibility is important too, so I think we must stay on C++14 ;)

**Comment by soroshsabz** [comment]
> I finally pass two test of three, so I wait to Microsoft VCPKG PR to continue work for third test. :)

At last vcpkg PR merged to master, so I can resolve mac problem, :)

**Comment by soroshsabz** [comment]
@fnc12 At last travis is passed all test :) so please review it :)

thanks

**Comment by fnc12** [comment]
thanks

**Pull request merged**